tuf/api/metadata.py

```diff
@@ -670,12 +671,20 @@ def _verify_hashes(
         """Verifies that the hash of 'data' matches 'expected_hashes'"""
         is_bytes = isinstance(data, bytes)
         for algo, exp_hash in expected_hashes.items():
-            if is_bytes:
-                digest_object = sslib_hash.digest(algo)
-                digest_object.update(data)
-            else:
-                # if data is not bytes, assume it is a file object
-                digest_object = sslib_hash.digest_fileobject(data, algo)
+            try:
+                if is_bytes:
+                    digest_object = sslib_hash.digest(algo)
+                    digest_object.update(data)
+                else:
+                    # if data is not bytes, assume it is a file object
+                    digest_object = sslib_hash.digest_fileobject(data, algo)
+            except (
+                sslib_exceptions.UnsupportedAlgorithmError,
+                sslib_exceptions.FormatError,
+            ) as e:
+                raise exceptions.UnsupportedAlgorithmError(
```

Comment:
assuming we go with this, it needs to be documented in at least the public methods